### PR TITLE
Update ModSDK

### PR DESCRIPTION
**Pull request by penev92**
Updates the ModSDK scripts, VSCode configuration, GitHub Actions workflows, etc. to match the current (albeit WIP) version of the ModSDK.

This is all based on (a copy of) my latest work on https://github.com/pchote/OpenRAModSDK/pull/2

### CODE REVIEW

**Comment by penev92** [comment]
Code style violations 😢 
Will be simplest to remove the changes to `.editorconfig`.

**Review by pchote — COMMENTED** [review]
Keep the editorconfig - i'll fix the code instead.

**Comment by pchote** [comment]
I have pushed fixes to the master branch.  Can you please rebase?

File: launch-dedicated.cmd [hunk 1]
```diff
@@ -3,9 +3,11 @@
 @echo on
 
 set Name="Dedicated Server"
+set Mod=cnc
```
[the remark this answers is not in the document]
Comment (penev92) [reply launch-dedicated.cmd:6]: It is. You did put "cnc" there.

[the remark this answers is not in the document]
Comment (penev92) [reply launch-dedicated.cmd:6]: If you mean it should be reading it from there - I think `launch-dedicated.cmd` is the odd one out of all 4 launch scripts which doesn't.

**Comment by penev92** [comment]
Rebased and fixed some more code style violations for you.

**Comment by pchote** [comment]
Can you please also bump the workflow action versions, as is currently being discussed in https://github.com/OpenRA/OpenRA/pull/20589 ?

**Comment by penev92** [comment]
Pushed with updated GitHub Actions versions.

**Review by pchote — APPROVED** [review]

File: .vscode/tasks.json [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+					"ENGINE_DIR": "${workspaceRoot}/engine",
+					"MOD_SEARCH_PATHS": "${workspaceRoot}/mods,${workspaceRoot}/engine/mods"
+				}
+			}
+		}
+	],
+	"inputs": [
+		{
+			"id": "modId",
+			"description": "ID of the mod to run",
```
Comment (Mailaender) [inline .vscode/tasks.json:27]: This repository only has one mod.

Comment (penev92) [reply .vscode/tasks.json:27]: You can also run commands on the base mods, although I don't know how relevant that would be for this mod in particular.

**Pull request merged**